File: calc.lua
function add(firstNumber, secondNumber)
   return firstNumber + secondNumber;
end

function subtract(firstNumber, secondNumber)
   return firstNumber - secondNumber;
end

function multiply(firstNumber, secondNumber)
   return firstNumber * secondNumber;
end

function divide(firstNumber, secondNumber)
   return firstNumber / secondNumber;
end

function inputNumber()
   io.write("Enter a number: ");
   number = io.read();
   return number;
end

while true do
   print("Select operation: ");
   print("1. Add\n2. Subtract\n3. Multiply\n4. Divide");
   io.write("Select from above: ");
   selectedOperation = io.read();
   if(selectedOperation == "0")then
      break;
   elseif(selectedOperation == "1")then
      print("The answer is:", add(inputNumber(), inputNumber()));
   elseif(selectedOperation == "2")then
      print("The answer is:", subtract(inputNumber(), inputNumber()));
   elseif(selectedOperation == "3")then
      print("The answer is:", multiply(inputNumber(), inputNumber()));
   elseif(selectedOperation == "4")then
      print("The answer is:", divide(inputNumber(), inputNumber()));
   else
      print("Wrong input");
   end
end

File: calc.go
package main

import (
  "fmt"
  "strconv"
  "os"
)
func Add(num1, num2 int) int {
        return num1 + num2
}

func Subtract(num1, num2 int) int {
        return num1 - num2
}

func Multiply(num1, num2 float64) float64 {
        return num1 * num2
}

func Divide(num1, num2 float64) float64 {
        return num1 / num2
}
func main() {
  getOperation:
        var operation string
  fmt.Print("Select operation: \n1. Add\n2. Subtract\n3. Multiply\n4. Divide\nSelect from above: ")
        fmt.Scanln(&operation)

  var num1 string
  fmt.Print("Enter the first number: ")
  fmt.Scanln(&num1)

  var num2 string
  fmt.Print("Enter the second number: ")
  fmt.Scanln(&num2)

  switch operation {
  case "1":
    var firstNumber int = stringToInt(num1)
    var secondNumber int = stringToInt(num2)
    fmt.Print("The answer is: ")
    fmt.Println(Add(firstNumber, secondNumber))
  case "2":
    var firstNumber int = stringToInt(num1)
    var secondNumber int = stringToInt(num2)
    fmt.Print("The answer is: ")
    fmt.Println(Subtract(firstNumber, secondNumber))

  case "3":
    var firstNumber float64 = stringToFloat64(num1)
    var secondNumber float64 = stringToFloat64(num2)
    fmt.Print("The answer is: ")
    fmt.Println(Multiply(firstNumber, secondNumber))

  case "4":
    var firstNumber float64 = stringToFloat64(num1)
    var secondNumber float64 = stringToFloat64(num2)
    fmt.Print("The answer is: ")
    fmt.Println(Divide(firstNumber, secondNumber))

  default:
    fmt.Println("Invalid input")
    goto getOperation
  }
}

func stringToInt(str string) int {
  i, err := strconv.Atoi(str)
  if err != nil {
    fmt.Println(err)
    os.Exit(2)
  }
  return i
}

func stringToFloat64(str string) float64 {
  f, err := strconv.ParseFloat(str, 64)
  if err != nil {
    fmt.Println(err)
    os.Exit(2)
  }
  return f
}
